### ISSUE TITLE
`scalaInstance.allJars` doesn't include scala-compiler.jar depending on Scala version

### ISSUE THREAD

**Issue by jastice** [sbt/sbt#3560]
Using any `scalaVersion` setting other than `"2.12.3"`, `scalaInstance.allJars` will be missing the `scala-compiler.jar`.

IntelliJ depends on this task to setup the compiler classpath when not using sbt directly.
Corresponding ticket: https://youtrack.jetbrains.com/issue/SCL-12591

## steps

* in `build.properties`: 
```
sbt.version = 1.0.2
```

* in `build.sbt`
```
scalaVersion := "2.12.1"
```

Run `consoleProject` on sbt shell:
```
> consoleProject

scala> scalaInstance.eval.allJars
res0: Array[java.io.File] = Array(/Users/jast/.ivy2/cache/org.scala-lang/scala-reflect/jars/scala-reflect-2.12.1.jar, /Users/jast/.ivy2/cache/org.scala-lang.modules/scala-xml_2.12/bundles/scala-xml_2.12-1.0.6.jar, /Users/jast/.ivy2/cache/jline/jline/jars/jline-2.14.1.jar)
```

Compare using `scalaVersion := "2.12.3"`

```
> consoleProject

scala> scalaInstance.eval.allJars
res2: Array[java.io.File] = Array(/Users/jast/.sbt/boot/scala-2.12.3/lib/jline.jar, /Users/jast/.sbt/boot/scala-2.12.3/lib/scala-compiler.jar, /Users/jast/.sbt/boot/scala-2.12.3/lib/scala-library.jar, /Users/jast/.sbt/boot/scala-2.12.3/lib/scala-reflect.jar, /Users/jast/.sbt/boot/scala-2.12.3/lib/scala-xml_2.12.jar)    
```

## notes

sbt version: 1.0.0 - 1.0.2

**Comment by eed3si9n** [comment]
Thank you for the fix.

**Comment by smarter** [comment]
Duplicate of https://github.com/sbt/sbt/issues/3491 ?

**Comment by eed3si9n** [comment]
Yup.

**Comment by eed3si9n** [comment]
Fixed in #3561
